cellxgene_schema_cli/tests/test_schema_compliance.py

```diff
@@ -1272,6 +1317,28 @@ def test_obsm_values_at_least_one_X(self):
             ["ERROR: At least one embedding in 'obsm' has to have a " "key with an 'X_' prefix."],
         )
 
+    def test_obsm_suffix_name_valid(self):
+        """
+        Suffix after X_ must be at least 1 character long
+        """
+        self.validator.adata.obsm["X_"] = self.validator.adata.obsm["X_umap"]
+        self.validator.validate_adata()
+        self.assertEqual(
+            self.validator.errors,
+            ["ERROR: Embedding key in 'adata.obsm' X_ must have a suffix at least one character long."],
+        )
+
+    def test_obsm_key_name_valid(self):
+        """
+        Embedding keys with whitespace are not valid
+        """
+        self.validator.adata.obsm["X_ umap"] = self.validator.adata.obsm["X_umap"]
+        self.validator.validate_adata()
+        self.assertEqual(
+            self.validator.errors,
+            ["ERROR: Embedding key X_ umap has whitespace in it, please remove it."],
+        )
+
     def test_obsm_shape(self):
```

Comment:
can you add a test for the other condition on this if conditional? i.e. must have same number of rows as X

Reply:
i updated for the case where the number of rows is not the same as the number of cells. it actually seems to fail when setting the key, not when running the validation. but the test should reflect that